### DISCUSSION TITLE
The behavior of the operating system's file system during file upload in MinIO？

### DISCUSSION THREAD

**Discussion by adababys** [minio/minio#19501] [Q&A] [answered]
For an erasure set of size 4, when uploading a 1MB file to MinIO, each disk needs to perform at least 2 directory creation operations and 2 file write operations, totaling 8 directory creation operations and 8 file write operations. If the uploaded file is small enough to be directly stored in the object's metadata file, then only 4 directory creation operations and 4 file write operations are needed. Compared to writing directly to the local file system, the number of file system operations required by MinIO for storing object data is increased by more than 8 times. Is this correct, and how can it be optimized?

**Comment by klauspost** [comment, marked as answer]
Correct overall. But if you write to a RAID 5/6 you will have write amplification to cover your redundancy - which is only "per host" - not per "cluster".

Redundancy is never free.
